imgix/urlbuilder.py

```diff
@@ -88,3 +114,75 @@ def create_url(self, path, params={}):
             params=params)
 
         return str(url_obj)
+
+    def create_srcset(self, path, params={}):
+        """
+        Create srcset attribute value with the supplied path and
+        `params` parameters dict.
+        Will generate a fixed-width DPR srcset if a width OR height and aspect
+        ratio are passed in as parameters. Otherwise will generate a srcset
+        with width-descriptor pairs.
+
+        Parameters
+        ----------
+        path : str
+        params : dict
+            Dictionary specifying URL parameters. Non-imgix parameters are
+            added to the URL unprocessed. For a complete list of imgix
+            supported parameters, visit https://docs.imgix.com/apis/url .
+            (default {})
+
+        Returns
+        -------
+        str
+            srcset attribute value
+        """
+
+        width = params['w'] if 'w' in params else None
+        height = params['h'] if 'h' in params else None
+        aspect_ratio = params['ar'] if 'ar' in params else None
+
+        if (width or (height and aspect_ratio)):
+            return self._build_srcset_DPR(path, params)
+        else:
+            return self._build_srcset_pairs(path, params)
+
+    def _target_widths(self):
+        resolutions = []
+        prev = 100
+
+        def ensure_even(n):
+            return 2 * round(n/2.0)
+
+        while prev <= SRCSET_MAX_SIZE:
+            resolutions.append(int(ensure_even(prev)))
+            prev *= 1 + (SRCSET_INCREMENT_PERCENTAGE / 100.0) * 2
+
+        resolutions.append(SRCSET_MAX_SIZE)
+        return resolutions
+
+    def _build_srcset_pairs(self, path, params=None):
+        if not params:
+            params = {}
+        srcset = ''
+        widths = self._target_widths()
+
+        for i in range(len(widths)):
+            current_width = widths[i]
+            current_params = params
+            current_params['w'] = current_width
+            srcset += self.create_url(path, current_params) \
+                + ' ' + str(current_width) + 'w,\n'
+
+        return srcset[0:-2]
+
+    def _build_srcset_DPR(self, path, params={}):
+        srcset = ''
+        target_ratios = [1, 2, 3, 4, 5]
```

Comment:
probably should pull this out into a constant `SRCSET_DPR_TARGET_RATIOS` or something and stick it up next to MAX_SIZE. Also `range(1, 6)` is slightly cleaner